akka-stream/src/main/scala/akka/stream/impl/ActorFlowMaterializerImpl.scala

```diff
@@ -74,7 +74,7 @@ private[akka] case class ActorFlowMaterializerImpl(override val settings: ActorF
             mat
 
           case stage: StageModule ⇒
-            val (processor, mat) = processorFor(stage, effectiveAttributes, calcSettings(effectiveAttributes)(settings))
+            val (processor, mat) = processorFor(stage, effectiveAttributes.and(OperationAttributes.materializationName(flowName)), calcSettings(effectiveAttributes)(settings))
```

Comment:
I am not sure about this. Aren't effectiveattributes already containing every relevant name?